tools/utils.ts

```diff
@@ -0,0 +1,71 @@
+import * as yargsParser from 'yargs-parser';
+import type * as Enquirer from 'enquirer';
+
+/**
+ * CLI prompts abstraction to trigger dynamic prompts within a generator
+ *
+ * @remarks
+ * - lazy loads enquirer only when needed making CLI programs faster to load/execute
+ *
+ * @param questions
+ */
+export async function prompt<T extends Record<string, unknown>>(questions: Parameters<Enquirer['prompt']>[0]) {
+  const EnquirerLazy = await await import('enquirer');
+
+  const response = await EnquirerLazy.prompt<T>(questions);
+
+  return response;
+}
+
+/**
+ * Determine if manual dynamic prompts should be enabled within generator
+ *
+ * @remarks
+ *
+ * This should be used IFF you need to setup manual dynamic prompts within your generator.
```

Comment:
```suggestion
 * This should be used if you need to setup manual dynamic prompts within your generator.
```

Reply:
I used `IFF` on purpose here https://en.wikipedia.org/wiki/If_and_only_if

Reply:
Ahh, then I suggest we go with verbose here to avoid confusion

```suggestion
 * This should be used if, and only if, you need to setup manual dynamic prompts within your generator.
```